src/modules/private.ts

```diff
@@ -707,6 +710,68 @@ export default class Private {
     );
   }
 
+  /**
+   * @description get trading rewards for a user for a given epoch
+   *
+   * @param {
+   * @epoch number to request rewards data for
+   * }
+   */
+  getTradingRewards(
+    params: {
+      epoch?: number,
+    },
+    genericParams: GenericParams = {},
+  ): Promise<{ tradingRewards: TradingRewardsResponseObject }> {
+    return this._get(
+      'rewards/weight',
+      {
+        ...params,
+        ...genericParams,
+      },
+    );
+  }
+
+  /**
+   * @description get liquidity provider rewards for a user for a given epoch
+   *
+   * @param {
+   * @epoch number to request rewards data for
+   * }
+   */
+  getLiquidityProviderRewards(
+    params: {
+      epoch?: number,
+    },
+    genericParams: GenericParams = {},
+  ): Promise<{ liquidityRewards: LiquidityProviderRewardsResponseObject }> {
+    return this._get(
+      'rewards/liquidity',
+      {
+        ...params,
+        ...genericParams,
+      },
+    );
+  }
+
+  /**
+   * @description get retoractive mining rewards for a user for a given epoch
+   *
+   * @param {
+   * @epoch number to request rewards data for
```

Comment:
any way to indicate that this is optional?

Reply:
ah this endpoint actually doesn't need the epoch, so i removed it as a param